### PR TITLE
Add: End 2 end test to InnerBlocks allowed block restrictions

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
## Description
This PR adds an end to end tests to the global inserter.
It verifies that:
- The inserter opens and shows all the blocks by default.
- On "Media & Text" block, the inserter items are restricted to a small set.
- Inside columns block, no item appears on the inserter.

Catches the bug being addressed in https://github.com/WordPress/gutenberg/pull/14020. So currently the tests fail.

### CODE REVIEW

File: packages/e2e-tests/specs/inserter.test.js
Comment (gziolo) [inline packages/e2e-tests/specs/inserter.test.js]: This test is very specific to the Media & Text block. It's fine to start with but I would personally prefer to see this kind of tests inside `blocks/media-text.spec.js` file. It's faster to start with such verification but ideally, we use some custom block registered for testing purpose.

File: packages/e2e-tests/specs/inserter.test.js
Comment (gziolo) [inline packages/e2e-tests/specs/inserter.test.js]: The same applies as above. Those two test cases should be documented with the explanation of why those blocks are used and what characteristics they have.

Example, if we would change the behavior of Columns blocks to allow some blocks we should rather remove this test from the test suite rather than update it.

File: packages/e2e-test-utils/src/open-global-inserter.js
Comment (gziolo) [inline packages/e2e-test-utils/src/open-global-inserter.js]:
```suggestion
 * Opens the global block inserter.
```

File: packages/e2e-test-utils/src/open-global-inserter.js
Comment (gziolo) [inline packages/e2e-test-utils/src/open-global-inserter.js]: Can we make it more specific?
```suggestion
export async function openGlobalBlockInserter() {
```

**Comment by jorgefilipecosta** [comment]
Hi @gziolo,
Thank you for the review, this PR was updated taking into account your feedback.
We now have block specific tests and a test that specifically tests the allowed block restrictions using a custom block.

File: packages/e2e-tests/plugins/inner-blocks-allowed-blocks.php
Comment (gziolo) [inline packages/e2e-tests/plugins/inner-blocks-allowed-blocks.php]: `wp-components` and `wp-hooks` are never used so can be safely removed.

File: packages/e2e-tests/plugins/inner-blocks-allowed-blocks.php
Comment (gziolo) [inline packages/e2e-tests/plugins/inner-blocks-allowed-blocks.php]: This name was copied over from another file, it should be updated.

File: packages/e2e-tests/plugins/inner-blocks-allowed-blocks.php
Comment (gziolo) [inline packages/e2e-tests/plugins/inner-blocks-allowed-blocks.php]: The same issue with the function name, needs to be synced with the name of the file/plugin.

File: packages/e2e-tests/specs/blocks/collumns.test.js
Comment (gziolo) [inline packages/e2e-tests/specs/blocks/collumns.test.js]: Typo here:
```suggestion
describe( 'Columns', () => {
```

and in the name of the file.

File: packages/e2e-test-utils/src/index.js
Comment (gziolo) [inline packages/e2e-test-utils/src/index.js]: Can we also update CHANGELOG file? This package is public and is already being explored by external projects. This is going to be minor version bump as we add new features 🎉

**Review by gziolo — APPROVED** [review]
Awesome work @jorgefilipecosta. I thought you would handle my comments in a separate PR :)

This looks great, I left a couple of minor things to address before merging. Overall, I'm impressed with how e2e testing infrastructure and coverage grows thanks to your numerous contributions. 🥇 

BTW, I ✅ this PR, if you feel you need another review when typos are fixed, let me know.

**Comment by jorgefilipecosta** [comment]
Thank you for the feedback @gziolo, it was applied before the merge.

**Comment by aduth** [comment]
I received an intermittent failure from these tests in #14128

```
Summary of all failing tests
FAIL specs/plugins/inner-blocks-allowed-blocks.test.js (8.254s)
  ● Allowed Blocks Setting on InnerBlocks  › allows all blocks if the allowed blocks setting was not set
    Node is detached from document
      at ElementHandle._scrollIntoViewIfNeeded (../../node_modules/puppeteer/lib/ElementHandle.js:75:13)
```

**Comment by aduth** [comment]
I note #14450 failed as well due to the same error.

**Comment by aduth** [comment]
Specifically, the error occurs here in Puppeteer. `_scrollIntoViewIfNeeded` is called by various interactions like clicking.

https://github.com/GoogleChrome/puppeteer/blob/v1.6.2/lib/ElementHandle.js#L60

Also worth noting the file doesn't exist in more recent versions of Puppeteer (unsure if that may have an impact).

**Pull request merged**
